centinel/primitives/tls.py

```diff
@@ -22,6 +23,11 @@ def get_fingerprint(host, port=443, external=None):
         tls_error = str(exp)
         exception = exp
 
+    # this comes out as unicode, but m2crypto breaks if it gets
+    # something other than a string, so convert to ascii
+    if type(cert) == unicode or test:
+        cert = cert.encode('ascii', 'ignore')
```

Comment:
`'ignore'` will ignore any error when changing encoding. That's a bad idea. If there is an exception, it should be caught within `try: except`.